### PR TITLE
chore: add  integration test for cluster mode.

### PR DESCRIPTION
**Pull request by ZuLiangWang**
# Which issue does this PR close?

Closes #

# Rationale for this change
Add integration test for cluster mode.

# What changes are included in this PR?
* Add docker-compose file to start CeresDB and CeresMeta cluster.
* Add CI for cluster mode.
* Update `database.rs` to support cluster mode `sqlness`.

# Are there any user-facing changes?
None.

# How does this change test
Test workflow in my fork repository. https://github.com/ZuLiangWang/ceresdb/actions

### CODE REVIEW

File: docker/example-standalone.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+etcd-start-timeout-ms = 30000
```
Comment (ShiKaiWi) [inline docker/example-standalone.toml:1]: Rename this file to `ceresmeta.toml`?

File: docker/example-cluster-1.toml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+[server]
```
Comment (ShiKaiWi) [inline docker/example-cluster-1.toml:1]: Rename this file to `ceresdb-0.toml`?

File: integration_tests/Cargo.toml [hunk 3]
```diff
@@ -11,3 +11,5 @@ async-trait = "0.1"
 ceresdb-client-rs = { git = "https://github.com/CeresDB/ceresdb-client-rs.git", rev = "5fbd1a1526c3ddd25bb1f38f63f869c892052f7c" }
 sqlness = "0.3"
 tokio = { workspace = true }
+sql = { workspace = true }
+sqlparser = { workspace = true }
```
Comment (ShiKaiWi) [inline integration_tests/Cargo.toml:15]: Missing one newline below.

File: integration_tests/src/database.rs [hunk 4]
```diff
@@ -70,18 +80,95 @@ impl CeresDB {
...
-            sql: query,
+
+        let mut statements = Parser::parse_sql(query.as_str()).unwrap();
+        // if statements.clone().len() != 1 {
+        //     format!(
+        //         "Failed to parse sql, statements length:{:?}, sql:{:?} ",
+        //         statements.clone().len(),
+        //         sql
+        //     )
+        // }
```
Comment (ShiKaiWi) [inline integration_tests/src/database.rs:101]: Remove these unused codes.

File: integration_tests/src/database.rs [hunk 5]
```diff
@@ -92,4 +130,74 @@ impl CeresDB {
...
+                    into: _,
+                    table_name,
+                    columns: _,
+                    overwrite: _,
+                    source: _,
+                    partitioned: _,
+                    after_columns: _,
+                    table: _,
+                    on: _,
+                    returning: _,
```
Comment (chunshao90) [inline integration_tests/src/database.rs:150]:
```suggestion
                SqlStatement::Insert {
                    table_name,
                    ..
```

File: integration_tests/src/database.rs [hunk 6]
```diff
@@ -92,4 +130,74 @@ impl CeresDB {
...
+                } => Some(TableName::from(table_name).to_string()),
+                SqlStatement::Explain {
+                    describe_alias: _,
+                    analyze: _,
+                    verbose: _,
+                    statement,
+                    format: _,
```
Comment (chunshao90) [inline integration_tests/src/database.rs:157]: Same above.

File: integration_tests/src/main.rs [hunk 7]
```diff
@@ -17,12 +20,22 @@ pub struct CeresDBController;
...
     type DB = CeresDB;
 
-    async fn start(&self, _env: &str, config: Option<&Path>) -> Self::DB {
-        CeresDB::new(config)
+    async fn start(&self, env: &str, config: Option<&Path>) -> Self::DB {
+        println!("start with env {env}");
+        match env {
+            "local" => CeresDB::new(config, DeployMode::Standalone),
+            "cluster" => CeresDB::new(config, DeployMode::Cluster),
+            _ => panic!("invalid env"),
```
Comment (chunshao90) [inline integration_tests/src/main.rs:28]: Print the invalid env.

**Comment by jiacai2050** [comment]
Don't need to duplicate test cases files, add a `common` dir, and soft link to both `local` and `cluster`.

This require this commit https://github.com/CeresDB/sqlness/pull/35, you can bump sqlness first.

File: integration_tests/src/database.rs [hunk 8]
```diff
@@ -45,40 +57,83 @@ impl CeresDB {
...
         println!("Start {bin} with {config}...");
 
-        let server_process = Command::new(&bin)
-            .args(["--config", &config])
-            .stdout(stdout)
-            .stderr(stderr)
-            .spawn()
-            .unwrap_or_else(|_| panic!("Failed to start server at {bin:?}"));
+        match mode {
+            DeployMode::Standalone => {
```
Comment (jiacai2050) [inline integration_tests/src/database.rs:61]: Add start/stop_standalone and start/stop_cluster methods

File: integration_tests/src/database.rs [hunk 9]
```diff
@@ -92,4 +147,74 @@ impl CeresDB {
...
         })
     }
+
+    fn parse_table_name(query: &str) -> Option<String> {
+        let statements = Parser::parse_sql(query).unwrap();
+
+        match &statements[0] {
+            Statement::Standard(s) => match *s.clone() {
+                SqlStatement::Insert {
+                    table_name,
```
Comment (jiacai2050) [inline integration_tests/src/database.rs:157]:
```
Insert {
  table_name
  ...
}
```

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -134,18 +139,24 @@ jobs:
...
+        uses: actions/checkout@v3
+        with:
+          repository: CeresDB/ceresmeta
+          ref: refs/heads/main
+      - name: Build CeresMeta Docker image
+        run: docker build -t ${CERESMETA_IMAGE_NAME} .
+      - name: Add CeresDB0 host
+        run: |
+          sudo echo "127.0.0.1 ceresdb0" | sudo tee -a /etc/hosts
+      - name: Add CeresDB1 host
```
Comment (jiacai2050) [inline .github/workflows/ci.yml:154]: Combine with last command.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**